### PR TITLE
MRG: deprecated ConcatenateChannels and renamed EpochVectorizer

### PR DESCRIPTION
**Pull request by trachelr**
@agramfort @dengemann This PR addresses issue #2313 
Is it fine to import ConcatenateChannels as EpochVectorizer in the __init__ ??
or I should copy the code of ConcatenateChannels and name it class EpochVectorizer?

### CODE REVIEW

**Comment by trachelr** [comment]
doc test don't pass because of deprecated call in the init...
so I'm gonna copy paste and rename ConcatenateChannels

File: doc/whats_new.rst [hunk 1]
```diff
@@ -45,6 +45,8 @@ Changelog
 
     - Add support to append new channels to an object from a list of other objects by `Chris Holdgraf`_
 
+    - Deprecated :class: ConcatenateChannels and replaced by :class: EpochVectorizer by `Romain Trachel`_
```
Comment (agramfort) [inline doc/whats_new.rst]: EpochVectorizer -> EpochsVectorizer

File: mne/decoding/tests/test_transformer.py [hunk 2]
```diff
@@ -119,8 +119,8 @@ def test_psdestimator():
     assert_raises(ValueError, psd.transform, epochs, y)
 
 
-def test_concatenatechannels():
-    """Test methods of ConcatenateChannels
+def test_epochvectorizer():
```
Comment (agramfort) [inline mne/decoding/tests/test_transformer.py]: test_epochvectorizer -> test_epochs_vectorizer

**Comment by agramfort** [comment]
besides LGTM

**Comment by dengemann** [comment]
Oops. Sorry guys, I merged to quick.

File: doc/whats_new.rst (hunk 1 above)
Comment (Eric89GXL) [inline doc/whats_new.rst]: It also needs to be in backticks to render properly

File: mne/decoding/transformer.py [hunk 3]
```diff
@@ -147,13 +147,13 @@ def inverse_transform(self, epochs_data, y=None):
...
 
+@deprecated("Class 'ConcatenateChannels' has been renamed to "
+            "'EpochsVectorizer' and will be removed in release 0.11.")
 class ConcatenateChannels(TransformerMixin):
     """Concatenates data from different channels into a single feature vector
 
     Parameters
     ----------
-    info : instance of Info
-        The measurement info.
```
Comment (trachelr) [inline mne/decoding/transformer.py]: it didn't pass doc tests because of deprecated.
I found that removing parameters doc fix this issue

**Comment by trachelr** [comment]
ready to merge, if doc test and travis are happy

File: mne/decoding/transformer.py (hunk 3 above)
Comment (Eric89GXL) [inline mne/decoding/transformer.py]: Don't fix it this way, fix it by adding the function or class name here:

https://github.com/mne-tools/mne-python/blob/master/mne/tests/test_docstring_parameters.py#L64

**Comment by agramfort** [comment]
ping me when you addressed @Eric89GXL comments

**Comment by trachelr** [comment]
done! thanks @Eric89GXL

**Comment by trachelr** [comment]
Yep because I get this error from test_docstring_parameters
AssertionError: 
mne.utils.ConcatenateChannels.__init__ arg mismatch: ['info']

File: mne/decoding/transformer.py [hunk 4]
```diff
@@ -147,6 +147,8 @@ def inverse_transform(self, epochs_data, y=None):
         return X
 
 
+@deprecated("Class 'ConcatenateChannels' has been renamed to "
+            "'EpochsVectorizer' and will be removed in release 0.11.")
 class ConcatenateChannels(TransformerMixin):
```
Comment (agramfort) [inline mne/decoding/transformer.py]: can't you just do:

```
@deprecated("Class 'ConcatenateChannels' has been renamed to  ...")
class ConcatenateChannels(EpochsVectorizer):
     pass
```

in order to remove the old code of ConcatenateChannels?

Comment (trachelr) [inline mne/decoding/transformer.py]: Yep, @Eric89GXL fix removed. 
It seems to pass docstring test.
Thanks @agramfort !

**Comment by agramfort** [comment]
perfect !

let's merge when travis is happy !

**Comment by Eric89GXL** [comment]
@trachelr for future reference you could have just added `ConcatenateChannels` to the list of excludes and it would have worked, too (it matches substrings)

**Comment by trachelr** [comment]
Got it @Eric89GXL !
Thanks

**Comment by agramfort** [comment]
thanks @trachelr

**Comment by mainakjas** [comment]
@trachelr the deprecations must go to the API section of `whats_new`. I just noticed this while rebasing. I'll make a fix in my PR but just so that you know for the future :)

**Comment by trachelr** [comment]
Ok! thanks @mainakjas

**Pull request merged**